### DISCUSSION TITLE
sanhook iscsi was register, but device is not present in the system after boot

### DISCUSSION THREAD

**Discussion by moemoe0spb** [ipxe/ipxe#812] [General]
iPxe ver. 1.21
I want to install Centos on iScsi root device.
ISCSI work on TrueNas 11. Target works with any OS. 

mac-xx-xx.xx.xx.ipxe:
sanhook --drive 0x81 iscsi:192.168.78.26:::f0:iqn.2005-10.org.freenas.ctl:hdvm001

set base http://192.168.78.23/boot/centos8
kernel ${base}/images/pxeboot/vmlinuz initrd=initrd.img inst.repo=${base} edd=off
initrd ${base}/images/pxeboot/initrd.img
boot

Device is registered:
<img width="731" alt="image" src="https://user-images.githubusercontent.com/118636514/202874957-a6b6bb62-32fd-42fe-8be5-3afa545bd44a.png">

On Centos 8 installer I have:
<img width="613" alt="image" src="https://user-images.githubusercontent.com/118636514/202872668-39a49127-9626-4c59-ad69-f6411b5931cc.png">

How I can attach iscsi root?

**Comment by NiKiZe** [comment]
Not an issue in iPXE, feel free to use discussions, don't open issues unless there is clear bugs with iPXE.

You will have to make sure that whatever system you are using does have iSCSI and in this case iBFT support.
